src/commonMain/kotlin/com/charleskorn/kaml/YamlNode.kt

```diff
@@ -248,3 +250,22 @@ public data class YamlTaggedNode(val tag: String, val innerNode: YamlNode) : Yam
 
     override fun toString(): String = "tagged '$tag': $innerNode"
 }
+
+public val YamlNode.yamlScalar: YamlScalar
+    get() = this as? YamlScalar ?: error(this, "YamlScalar")
+
+public val YamlNode.yamlNull: YamlNull
+    get() = this as? YamlNull ?: error(this, "YamlNull")
+
+public val YamlNode.yamlList: YamlList
+    get() = this as? YamlList ?: error(this, "YamlList")
+
+public val YamlNode.yamlMap: YamlMap
+    get() = this as? YamlMap ?: error(this, "YamlMap")
+
+public val YamlNode.yamlTaggedNode: YamlTaggedNode
+    get() = this as? YamlTaggedNode ?: error(this, "YamlTaggedNode")
+
+private fun error(node: YamlNode, expectedType: String): Nothing {
+    throw IncorrectTypeException("Element is not $expectedType", node.path)
```

Comment:
It would be good to include the actual type of the node in addition to what was expected (eg. "Expected element to be a list, but is a scalar")

Reply:
Done.